src/bitcoin/block.h

```diff
@@ -77,26 +173,33 @@ class CBlock : public CBlockHeader {
     ADD_SERIALIZE_METHODS
 
     template <typename Stream, typename Operation>
-    inline void SerializationOp(Stream &s, Operation ser_action) {
+        inline void SerializationOp(Stream &s, Operation ser_action) {
         READWRITEAS(CBlockHeader, *this);
+        READWRITE(vMetadata);
         READWRITE(vtx);
     }
 
     void SetNull() {
         CBlockHeader::SetNull();
         vtx.clear();
+        vMetadata.clear();
         fChecked = false;
     }
 
     CBlockHeader GetBlockHeader() const {
-        CBlockHeader block;
-        block.nVersion = nVersion;
-        block.hashPrevBlock = hashPrevBlock;
-        block.hashMerkleRoot = hashMerkleRoot;
-        block.nTime = nTime;
-        block.nBits = nBits;
-        block.nNonce = nNonce;
-        return block;
+        CBlockHeader header;
+        header.hashPrevBlock = hashPrevBlock;
+        header.nBits = nBits;
+        header.vTime = vTime;
+        header.nReserved = nReserved;
+        header.nNonce = nNonce;
+        header.nHeaderVersion = nHeaderVersion;
+        header.vSize = vSize;
+        header.nHeight = nHeight;
+        header.hashEpochBlock = hashEpochBlock;
+        header.hashMerkleRoot = hashMerkleRoot;
+        header.hashExtendedMetadata = hashExtendedMetadata;
+        return header;
```

Comment:
It's not clear to me what even the point of this method is in the original bitcoin code.. but this would work:

```suggestion
    return *this;
```